### PR TITLE
Fixes for [CE-4007], [CE-4016], [CE-4017]

### PR DESCRIPTION
**Pull request by CraigLn**
Bumped to next major version due to non-passive changes relating to error handling.  

See changelog for more detailed breakdown of changes.

### CODE REVIEW

File: .pubnub.yml [hunk 1]
```diff
@@ -1,9 +1,33 @@
 ---
 name: swift
 scm: github.com/pubnub/swift
-version: "1.2.1"
+version: "2.0.0"
```
Comment (davidnub) [inline .pubnub.yml]: Must be a pretty big change then.

File: Sources/PubNub/Events/Session/SessionStream.swift [hunk 2]
```diff
@@ -50,8 +50,8 @@ public protocol SessionStream: EventStream {
   func emitRequest(_ request: Request, didFailToMutate initialURLRequest: URLRequest, with error: Error)
 
   // HTTPResponse
-  func emitDidDecode(_ initialResponse: Response<Data>)
-  func emitFailedToDecode(_ initialResponse: Response<Data>, with error: Error)
+//  func emitDidDecode(_ initialResponse: Response<Data>)
```
Comment (davidnub) [inline Sources/PubNub/Events/Session/SessionStream.swift]: Remove or comment out?

File: Sources/PubNub/Events/Session/SessionStream.swift [hunk 3]
```diff
@@ -78,8 +78,8 @@ public extension SessionStream {
   func emitRequest(_: Request, didMutate _: URLRequest, to _: URLRequest) { /* no-op */ }
   func emitRequest(_: Request, didFailToMutate _: URLRequest, with _: Error) { /* no-op */ }
 
-  func emitDidDecode(_: Response<Data>) { /* no-op */ }
-  func emitFailedToDecode(_: Response<Data>, with _: Error) { /* no-op */ }
+//  func emitDidDecode(_: Response<Data>) { /* no-op */ }
```
Comment (davidnub) [inline Sources/PubNub/Events/Session/SessionStream.swift]: Remove or comment out?

File: Sources/PubNub/Events/Session/SessionStream.swift [hunk 4]
```diff
@@ -167,13 +167,13 @@ public final class MultiplexSessionStream: SessionStream, Hashable {
...
   }
 
-  public func emitDidDecode(_ initialResponse: Response<Data>) {
-    performEvent { $0.emitDidDecode(initialResponse) }
-  }
-
-  public func emitFailedToDecode(_ response: Response<Data>, with error: Error) {
-    performEvent { $0.emitFailedToDecode(response, with: error) }
-  }
+//  public func emitDidDecode(_ initialResponse: Response<Data>) {
```
Comment (davidnub) [inline Sources/PubNub/Events/Session/SessionStream.swift]: Remove or comment out?

File: Sources/PubNub/Events/Session/SessionStream.swift [hunk 5]
```diff
@@ -219,8 +219,8 @@ final class SessionListener: SessionStream, Hashable {
   var didMutateRequest: ((Request, URLRequest, URLRequest) -> Void)?
   var didFailToMutateRequest: ((Request, URLRequest, Error) -> Void)?
 
-  var didDecodeResponse: ((Response<Data>) -> Void)?
-  var failedToDecodeResponse: ((Response<Data>, Error) -> Void)?
+//  var didDecodeResponse: ((Response<Data>) -> Void)?
```
Comment (davidnub) [inline Sources/PubNub/Events/Session/SessionStream.swift]: Remove or comment out?

File: Sources/PubNub/Events/Session/SessionStream.swift [hunk 6]
```diff
@@ -283,13 +283,13 @@ final class SessionListener: SessionStream, Hashable {
...
 
   // HTTPResponse
-  func emitDidDecode(_ initialResponse: Response<Data>) {
-    queue.async { self.didDecodeResponse?(initialResponse) }
-  }
-
-  func emitFailedToDecode(_ initialResponse: Response<Data>, with error: Error) {
-    queue.async { self.failedToDecodeResponse?(initialResponse, error) }
-  }
+//  func emitDidDecode(_ initialResponse: Response<Data>) {
```
Comment (davidnub) [inline Sources/PubNub/Events/Session/SessionStream.swift]: Remove or comment out?

File: .pubnub.yml (hunk 1 above)
Comment (CraigLn) [reply .pubnub.yml]: I wasn't sure if we're following semantic versioning or doing feature based versioning.  It's a semantic major version increase, but feature wise it doesn't add anything special.

Comment (davidnub) [reply .pubnub.yml]: I'm fine with this, we should be following semantic versioning, it's just not "officially" implemented yet. Don't forget to create a gist or whatever for documentation changes.

File: Sources/PubNub/Events/Session/SessionStream.swift (hunk 2 above)
Comment (CraigLn) [reply Sources/PubNub/Events/Session/SessionStream.swift]: Removed

File: Sources/PubNub/Events/Session/SessionStream.swift (hunk 4 above)
Comment (CraigLn) [reply Sources/PubNub/Events/Session/SessionStream.swift]: Removed

File: Sources/PubNub/Events/Session/SessionStream.swift (hunk 6 above)
Comment (CraigLn) [reply Sources/PubNub/Events/Session/SessionStream.swift]: Removed

File: .pubnub.yml (hunk 1 above)
Comment (CraigLn) [reply .pubnub.yml]: Sounds good; working on the docs update.

**Pull request merged**